pkg/downloader/custom_github_detector.go

```diff
@@ -0,0 +1,110 @@
+package downloader
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/charmbracelet/log"
+	"github.com/cloudposse/atmos/pkg/schema"
+)
+
+// Predefined errors for better reuse and static analysis.
+var (
+	ErrInvalidGitHubURL = errors.New("invalid GitHub URL")
+	ErrURLEmpty         = errors.New("source URL is empty")
+	ErrFailedParseURL   = errors.New("failed to parse URL")
+)
+
+type EnvProvider func(key string) string
+
+// Logger interface allows logging abstraction for testing.
+type Logger interface {
+	Debug(any, ...any)
+}
+
+// customGitHubDetector detects and modifies GitHub URLs.
+type customGitHubDetector struct {
+	AtmosConfig *schema.AtmosConfiguration
+	GetEnv      EnvProvider
+	Log         Logger
+}
+
+// NewCustomGitHubDetector initializes a new detector with default dependencies.
+func NewCustomGitHubDetector(config *schema.AtmosConfiguration) *customGitHubDetector {
+	return &customGitHubDetector{
+		AtmosConfig: config,
+		GetEnv:      os.Getenv,
+		Log:         log.Default(),
```

Comment:
We shouldn't need to do this as we have a global logger already.

Reply:
Done